### ISSUE TITLE
GPS all of a sudden not working home arrow gone and flashing

### ISSUE THREAD

**Issue by wingflyer101** [iNavFlight/inav#5744]
using a f405 and bn880 GPS everything is fine and has been working fine apart from my home arrow blinking then when I arm I'm left with a small line that blinks so can't tell were home is I've been ripping my hair out trying to fix this, I have a 3d fix and 15-16 sats. I have tried another gps with exactly the same problem.

**Comment by AnandNL** [comment]
during flight or on the ground? i have had in flight that de gps stopped working, the runcam split turned out to be the problem jamming the GPS 

[https://www.youtube.com/watch?v=JpVrqozWXbY](url)

[https://youtu.be/jo6tbFnQrfI?t=36](url)

**Comment by DzikuVx** [comment]
the fact that GPS module was able to get the fix does not mean it will be able to keep it in a changing environment: electrical noise, RF nosie, weather conditions, etc.

**Comment by pandg-boop** [comment]
Hello. I reopen this thread because I have the same problem.

Inav 8, FC rcflying F4WSE mk 1.5.
I am using it on RC boat, everything works fine except that before arming direction to home indicator is flashing and after arming it turns into a flashing dash. Gps have a solid fix.

**Comment by sensei-hacker** [comment]
@pandg-boop  please attach your diff and a photo of your craft showing the mounting location of the GPS and what's around it.

Also, verify you have uploaded / switched to the proper INAV font for whatever video system you're using.

**Comment by pandg-boop** [comment]
> [@pandg-boop](https://github.com/pandg-boop) please attach your diff and a photo of your craft showing the mounting location of the GPS and what's around it.
> 
> Also, verify you have uploaded / switched to the proper INAV font for whatever video system you're using.

Hi.

In the photo, the gps is in the bow of the boat, around 11 cm ahead of the flight controller (boat controller) that is below that brown cover, just ahead the mast.

Servos are behind the switch you see on the left, toward the stern, while analog VTX and SBUS rx are on top of the mast, around 75 cm above the hull.

Camera, with servo gimball, is installed, and perfectly working, at the stern.

Exact behaviour is that the return to home arrow flashes continuously since the power up, even after GPS fix and sensors calibrating is complete. 
As I arm the boat it turns into a flashing horizontal dash. I moved it on top of the screen but the behaviour is the same.

I have systems and warning messages on and everything seems to be ok after arming, the only problem is direction to home. Even distance from home is correctly updating while sailing.

Video format is set to "AUTO" (I am using PAL however) and the font I selected is impact mini, but i changed to other fonts without success.

Thanks for the attention.

[INAV_8.0.1_cli_20251114_160319.txt](https://github.com/user-attachments/files/23550074/INAV_8.0.1_cli_20251114_160319.txt)

![Image](https://github.com/user-attachments/assets/94cb7aa7-f17a-4757-993a-265ca489a79f)

**Comment by pandg-boop** [comment]
???

**Comment by sensei-hacker** [comment]
According to your diff, your compass type is set to None- you have no compass.  The BN880 does have a compass, so on the Configuration page try setting that to each of the 5883 options and see if your compass gets detected.

You'll also want to calibrate the compass on the Calibration page.

Using the location from the GPS, INAV can figure out that home is 600 meters to the east. But without a compass there's no way to know which way is east, and therefore no way to point an arrow east.

* For a fast-flying airplane with a tail, we can assume that if it's moving west, it's probably facing pretty much west. Not so will a sail boat - it could be drifting west with the current while facing south.

**Comment by pandg-boop** [comment]
Great!! Now I know where is the problem. I had understood, looking around the net, that compass was not necessary, I understood wrong.

Actually my gps antenna is this one:

https://it.aliexpress.com/item/1005005134658746.html?spm=a2g0o.order_list.order_list_main.72.41ae3696inz5S0&gatewayAdapt=glo2ita 

and I think it does not have the compass. If not, as I believe, I will buy the BN880, as you suggested.

Thank you so much. I'll let you know.


EDIT - Update: no compass in my gps, i've just ordered one with QMC5883. It will take a while

**Comment by pandg-boop** [comment]
GREAT! IT WORKED!
So: together with the new GPS antenna I ordered another FC, same model (RC Flying F4WSE), because I did not want to struggle with the one buried in the boat before being sure where the problem was.

Setted the QMC5883 magnetometer, calibrated everything, magnetometer icon blue, so everything was working.

 I went outside, because my laboratory is in the basement, tested, no improvement, even worst because INAV was saying " !NO HOME POSITION! " while arming, so not even distance to home, although GPS was seeing 12 satellites. 

Did some googling, found the HDOP variable, turned it on. And also set Failsafe to "do nothing" it was setted to RTH.
I couldn't test it because was late evening. 

This morning, in the daylight, BAM, everything working, HDOP 1.6, 17 satellites, Direction to home and distance perfectly reflected.

THANK YOU SENSEI (I have been  struggling with this for a while)
